src/utils/AutoDiscoveryUtils.js

```diff
@@ -197,20 +205,18 @@ export default class AutoDiscoveryUtils {
             preferredIdentityUrl = isResult["base_url"];
         } else if (isResult && isResult.state !== AutoDiscovery.PROMPT) {
             console.error("Error determining preferred identity server URL:", isResult);
-            if (!syntaxOnly || !AutoDiscoveryUtils.isLivelinessError(isResult.error)) {
+            if (isResult.state === AutoDiscovery.FAIL_ERROR) {
                 if (AutoDiscovery.ALL_ERRORS.indexOf(isResult.error) !== -1) {
                     throw newTranslatableError(isResult.error);
                 }
                 throw newTranslatableError(_td("Unexpected error resolving identity server configuration"));
             } // else the error is not related to syntax - continue anyways.
 
-            // rewrite homeserver error if we don't care about problems
-            if (syntaxOnly) {
-                hsResult.error = AutoDiscovery.ERROR_INVALID_IDENTITY_SERVER;
+            // rewrite homeserver error since we don't care about problems
```

Comment:
we do care about problems - this `syntaxOnly` flag is used during app startup.

Reply:
I don't follow your comment here... The current change intends that we only reach this point if we got `FAIL_PROMPT`. If so, we want to record a warning with or without `syntaxOnly` mode.

Reply:
I think you're right, but i am concerned that the app won't load if there's a warning. If the app loads fine with no identity server (or a dead identity server) while the user is logged in, then this is fine.

Reply:
Yes, the app loads with a dead identity server while logged in.